src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -42,18 +48,43 @@ private static void validPaidSessionAmount(SessionAmount amount) {
         }
     }
 
-    public static Session createFreeSession(Long id, SessionPeriod period, SessionCoverImage coverImage, SessionStatus status) {
-        return new Session(id, period, coverImage, SessionFeeType.FREE, new SessionAmount(0L), 0, status);
+    public static Session freeSession(Long id, Long courseId, SessionPeriod period, SessionCoverImage coverImage, SessionStatus status) {
+        return new Session(id, courseId, period, coverImage, SessionFeeType.FREE, new SessionAmount(0L), 0, status);
+    }
+
+    public static Session from(SessionCreate sessionCreate) {
+        return new Session(
+                sessionCreate.getCourseId(),
+                new SessionPeriod(sessionCreate.getStartDate(), sessionCreate.getEndDate()),
+                new SessionCoverImage(sessionCreate.getFileSize(), sessionCreate.getFileType(), sessionCreate.getWidth(), sessionCreate.getHeight()),
+                sessionCreate.getAmount() == 0L ? SessionFeeType.FREE : SessionFeeType.PAID,
+                new SessionAmount(sessionCreate.getAmount()),
+                sessionCreate.getMaxPersonnel(),
+                SessionStatus.PREPARING);
+    }
+
+    public void open() {
+        if (!SessionStatus.PREPARING.equals(this.status)) {
```

Comment:
```suggestion
        if (status.isPreparing()) { // 예시
```
메시지를 보내도록 바꿔보면 어떨까요?